### ISSUE TITLE
Configuration Wizard: MailChimp input error identification

### ISSUE THREAD

**Issue by afercia** [Yoast/javascript#137]
Splitting this out from https://github.com/Yoast/yoast-components/issues/843
See also https://github.com/Yoast/design/issues/323

All the inputs that require a specific format should use an input validation mechanism and provide proper feedback to users.

In the Mailchimp email field:
- the entered value is validated, however the feedback is insufficient
- when there's an error, the input field is not clearly identified
- the item in error is not styled
- even if it was styled, color alone is not sufficient: it needs an additional shape
- the error text should be positioned above the field, see https://github.com/Yoast/wordpress-seo/pull/12456#issuecomment-475336422
- it is still possible to move to the next step even after entering a clearly invalid value

<img width="400" alt="52333373-61747e80-29fd-11e9-82d1-0a36b89127be" src="https://user-images.githubusercontent.com/1682452/53883832-7521fd80-401a-11e9-950d-9e597245885d.png">

<img width="660" alt="screenshot 2019-03-06 at 14 02 55" src="https://user-images.githubusercontent.com/1682452/53883833-7521fd80-401a-11e9-8007-a1f83d1d8cd5.png">

<img width="382" alt="screenshot 2019-03-06 at 14 15 27" src="https://user-images.githubusercontent.com/1682452/53883835-7521fd80-401a-11e9-8bf0-88767b6aeef6.png">

Good news:
- the visible message area is an ARIA live region so error (and success) messages are announced to assistive technologies
- there are proper suggestions for correction
